connectd/tor.c

```diff
@@ -41,6 +57,31 @@ struct connecting_socks {
 	struct connecting *connect;
 };
 
+static const char* socks5strerror(enum socks5_reply_code code)
+{
+	switch (code) {
+	case SOCKS5_REPLY_SUCCESS:
+		return "success";
+	case SOCKS5_REPLY_ERR_GENERAL:
+		return "general SOCKS server failure";
+	case SOCKS5_REPLY_ERR_NOT_ALLOWED:
+		return "connection not allowed by ruleset";
+	case SOCKS5_REPLY_ERR_NET_UNREACH:
+		return "network unreachable";
+	case SOCKS5_REPLY_ERR_HOST_UNREACH:
+		return "host unreachable";
+	case SOCKS5_REPLY_ERR_CONN_REFUSED:
+		return "connection refused";
+	case SOCKS5_REPLY_ERR_TTL_EXPIRED:
+		return "TTL expired";
+	case SOCKS5_REPLY_ERR_CMD_NOT_SUPP:
+		return "command not supported";
+	case SOCKS5_REPLY_ERR_ADDR_NOT_SUPP:
+		return "address type not supported";
+	}
+	return "other error";
```

Comment:
i'd also return the error number in this case.

Reply:
That would suggest passing in a `const tal_t *ctx` and using `tal_fmt` / `tal_strdup`, which I suppose ties well with the thinking of @Saibato .  Caller could pass in `tmpctx` for context.

Reply:
Done.